server/src/main/java/org/jboss/as/server/controller/git/ElytronClientSshdSessionFactory.java

```diff
@@ -0,0 +1,237 @@
+                this.keyPair = keyPairCredential.getKeyPair();
+            }
+
+        } catch (IOException  | UnsupportedCallbackException | URISyntaxException ex) {
+            ServerLogger.ROOT_LOGGER.failedToLoadSSHCredentials(ex, ex.getMessage());
+        }
+        return super.getSession(urIish, credentialsProvider, fs, tms);
+    }
+
+    @Override
+    protected KeyPasswordProvider createKeyPasswordProvider(CredentialsProvider provider) {
+        if (context == null) return super.createKeyPasswordProvider(provider);
+        URI uri = this.uri;
+        return new IdentityPasswordProvider(new CredentialsProvider() {
+           @Override
+           public boolean isInteractive() {
+               return false;
+           }
+
+           @Override
+           public boolean supports(CredentialItem... credentialItems) {
+               for (CredentialItem i : credentialItems) {
+                   if (!(i instanceof CredentialItem.Password)) {
+                       return false;
+                   }
+               }
+               return true;
+           }
+
+           @Override
+           public boolean get(URIish urIish, CredentialItem... credentialItems) throws UnsupportedCredentialItem {
+               try {
+                   AuthenticationConfiguration config = CLIENT.getAuthenticationConfiguration(uri, context);
+
+                   for (CredentialItem i : credentialItems) {
+                       if (i instanceof CredentialItem.Password) {
+                           //Only SSHCredential will require a passphrase to decrypt the private key, a KeyPairCredential already contains the decrypted private key
+                           CredentialCallback callback = new CredentialCallback(SSHCredential.class, null);
+                           Callback[] callbacks = {callback};
+                           CLIENT.getCallbackHandler(config).handle(callbacks);
+                           SSHCredential credential = null;
+                           if (callback != null && callback.getCredential() != null) credential = callback.getCredential().castAs(SSHCredential.class);
+                           char[] password = credential.getPassphrase().castAndApply(PasswordCredential.class, c -> c.getPassword().castAndApply(ClearPassword.class, ClearPassword::getPassword));
+                           ((CredentialItem.Password) i).setValue(password);
+                           continue;
+                       }
+                   }
+               } catch (IOException  | UnsupportedCallbackException ex) {
+                   throw new UnsupportedCredentialItem(urIish, ex.getMessage());
+               }
+
+               return true;
+           }
+       });
+    }
+
+    @Override
+    @NonNull
+    protected List<Path> getDefaultKnownHostsFiles(@NonNull File sshDir) {
+        return Arrays.asList(sshDir.toPath().resolve(this.knownHostsFile));
```

Comment:
This would be better as `Collections.singletonList(sshDir.toPath().resolve(this.knownHostsFile));`

Reply:
Fixed, thanks!